Makefile

```diff
@@ -40,14 +49,15 @@ xtest:
 ta:
 	$(q)$(MAKE) -C ta CROSS_COMPILE="$(CROSS_COMPILE_TA)" \
 			  q=$(q) \
-			  O=$(out-dir)/ta \
+			  O=$(out-dir) \
 			  $@
 
 .PHONY: clean
 ifneq ($(wildcard $(TA_DEV_KIT_DIR)/host_include/conf.mk),)
 clean:
 	$(q)$(MAKE) -C host/xtest O=$(out-dir)/xtest q=$(q) $@
-	$(q)$(MAKE) -C ta O=$(out-dir)/ta q=$(q) $@
+	cd host/xtest && rmdir $(O) 2> /dev/null; true
+	$(q)$(MAKE) -C ta O=$(out-dir) q=$(q) $@
```

Comment:
this was not needed for xtest so i discarded the equivalent change.
If you feel it is more coherent, then ok.